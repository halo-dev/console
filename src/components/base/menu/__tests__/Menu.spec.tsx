import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import { VMenu, VMenuItem } from "../index";

describe("Menu", () => {
  it("should render", () => {
    expect(VMenu).toBeDefined();
    expect(VMenuItem).toBeDefined();
    expect(mount(VMenu).html()).toMatchSnapshot();

    const wrapper = mount({
      render() {
        return (
          <VMenu>
            <VMenuItem key="1" title="Menu Item 1" />
            <VMenuItem key="2" title="Menu Item 2" />
          </VMenu>
        );
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should work with sub menus", function () {
    const wrapper = mount({
      setup() {
        return () => (
          <VMenu>
            <VMenuItem id="1" title="Menu Item 1" />
            <VMenuItem id="2" title="Menu Item 2" />
            <VMenuItem id="3" title="Menu Item 3" />
            <VMenuItem id="4" title="Menu Item 4" />
          </VMenu>
        );
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should work with openIds prop", function () {
    const wrapper = mount({
      setup() {
        return () => (
          <VMenu openIds={["3"]}>
            <VMenuItem id="1" title="Menu Item 1" />
            <VMenuItem id="2" title="Menu Item 2" />
            <VMenuItem id="3" title="Menu Item 3">
              <VMenuItem key="4" title="Menu Item 4" />
            </VMenuItem>
          </VMenu>
        );
      },
    });

    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.find(".sub-menu-items .menu-title").text()).contain(
      "Menu Item 4"
    );
  });
});
