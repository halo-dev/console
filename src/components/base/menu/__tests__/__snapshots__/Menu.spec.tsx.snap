// Vitest Snapshot v1

exports[`Menu > should render 1`] = `
"<div class=\\"menu-container w-full h-full\\">
  <ul></ul>
</div>"
`;

exports[`Menu > should render 2`] = `
"<div class=\\"menu-container w-full h-full\\">
  <ul>
    <li class=\\"menu-item\\">
      <div class=\\"menu-item-title\\">
        <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 1</span>
        <!--v-if-->
      </div>
      <transition-stub>
        <ul class=\\"sub-menu-items transition-all\\" style=\\"display: none;\\"></ul>
      </transition-stub>
    </li>
    <li class=\\"menu-item\\">
      <div class=\\"menu-item-title\\">
        <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 2</span>
        <!--v-if-->
      </div>
      <transition-stub>
        <ul class=\\"sub-menu-items transition-all\\" style=\\"display: none;\\"></ul>
      </transition-stub>
    </li>
  </ul>
</div>"
`;

exports[`Menu > should work with openIds prop 1`] = `
"<div class=\\"menu-container w-full h-full\\">
  <ul>
    <li class=\\"menu-item\\">
      <div class=\\"menu-item-title\\">
        <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 1</span>
        <!--v-if-->
      </div>
      <transition-stub>
        <ul class=\\"sub-menu-items transition-all\\" style=\\"display: none;\\"></ul>
      </transition-stub>
    </li>
    <li class=\\"menu-item\\">
      <div class=\\"menu-item-title\\">
        <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 2</span>
        <!--v-if-->
      </div>
      <transition-stub>
        <ul class=\\"sub-menu-items transition-all\\" style=\\"display: none;\\"></ul>
      </transition-stub>
    </li>
    <li class=\\"menu-item\\">
      <div class=\\"menu-item-title\\">
        <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 3</span><span class=\\"menu-icon-collapse self-center transition-all open\\"><svg preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\"><path fill=\\"currentColor\\" d=\\"m13.172 12l-4.95-4.95l1.414-1.414L16 12l-6.364 6.364l-1.414-1.414z\\"></path></svg></span>
      </div>
      <transition-stub>
        <ul class=\\"sub-menu-items transition-all\\">
          <li class=\\"menu-item\\">
            <div class=\\"menu-item-title\\">
              <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 4</span>
              <!--v-if-->
            </div>
            <transition-stub>
              <ul class=\\"sub-menu-items transition-all\\" style=\\"display: none;\\"></ul>
            </transition-stub>
          </li>
        </ul>
      </transition-stub>
    </li>
  </ul>
</div>"
`;

exports[`Menu > should work with sub menus 1`] = `
"<div class=\\"menu-container w-full h-full\\">
  <ul>
    <li class=\\"menu-item\\">
      <div class=\\"menu-item-title\\">
        <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 1</span>
        <!--v-if-->
      </div>
      <transition-stub>
        <ul class=\\"sub-menu-items transition-all\\" style=\\"display: none;\\"></ul>
      </transition-stub>
    </li>
    <li class=\\"menu-item\\">
      <div class=\\"menu-item-title\\">
        <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 2</span>
        <!--v-if-->
      </div>
      <transition-stub>
        <ul class=\\"sub-menu-items transition-all\\" style=\\"display: none;\\"></ul>
      </transition-stub>
    </li>
    <li class=\\"menu-item\\">
      <div class=\\"menu-item-title\\">
        <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 3</span>
        <!--v-if-->
      </div>
      <transition-stub>
        <ul class=\\"sub-menu-items transition-all\\" style=\\"display: none;\\"></ul>
      </transition-stub>
    </li>
    <li class=\\"menu-item\\">
      <div class=\\"menu-item-title\\">
        <!--v-if--><span class=\\"menu-title self-center flex-1\\">Menu Item 4</span>
        <!--v-if-->
      </div>
      <transition-stub>
        <ul class=\\"sub-menu-items transition-all\\" style=\\"display: none;\\"></ul>
      </transition-stub>
    </li>
  </ul>
</div>"
`;
